tests/integration/ambient/cacert_rotation_test.go

```diff
@@ -0,0 +1,125 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ambient
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+
+	"istio.io/istio/istioctl/pkg/util/configdump"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	kubetest "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/util/assert"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/security/util/cert"
+)
+
+func TestIntermediateCertificateRefresh(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.peer.cacert-rotation").
+		Run(func(t framework.TestContext) {
+			istioCfg := istio.DefaultConfigOrFail(t, t)
+			istioCtl := istioctl.NewOrFail(t, t, istioctl.Config{})
+			namespace.ClaimOrFail(t, t, istioCfg.SystemNamespace)
+
+			sa := apps.Captured[0].ServiceAccountName()
+
+			// we do not know which ztunnel instance is located on the node as the workload, so we need to check all of them initially
+			ztunnelPods, err := kubetest.NewPodFetch(t.AllClusters()[0], istioCfg.SystemNamespace, "app=ztunnel")()
+			assert.NoError(t, err)
+
+			originalPem, ztunnelPod, err := getWorkloadIntermediateCert(t, ztunnelPods, sa, istioCtl)
+			if err != nil {
+				t.Errorf("failed to get initial workload cert: %v", err)
+			}
+
+			// Update CA with new intermediate cert
+			if err := cert.CreateCustomCASecret(t,
+				"ca-cert-alt.pem", "ca-key-alt.pem",
+				"cert-chain-alt.pem", "root-cert-combined.pem"); err != nil {
+				t.Errorf("failed to update CA secret: %v", err)
+			}
+
+			newPem := waitForWorkloadCertUpdate(t, ztunnelPod, sa, istioCtl, originalPem)
```

Comment:
It will need to wait for more than 5 minutes

Reply:
In main_test.go, I specify `5m` as the ttl, but ztunnel uses a 0.5 ratio for cert renewal, so its renewed at about the two minute mark. I could make it shorter, but was concerned about excessive cert refreshes, especially because ztunnel still doesn't have a method for forgetting a cert if a workload no longer exists. So for instance there are CSRs coming in for bookinfo pods even after the bookinfo test runs and then deletes the corresponding namespace/workloads

Reply:
I think in the long run, the ideal way is to let istiod notify ztunnel that root cert has changed like sidecar does now.

Reply:
agreed